contracts/security-token/ERC865.sol

```diff
@@ -0,0 +1,103 @@
+pragma solidity ^0.4.18;
+
+import "./CheckpointToken.sol";
+
+/* Largely copied from https://github.com/OpenZeppelin/openzeppelin-solidity/pull/741/files */
+
+contract ERC865 is CheckpointToken {
+  mapping(bytes => bool) signatures;
+
+  event TransferPreSigned(address indexed from, address indexed to, address indexed delegate, uint256 amount, uint256 fee);
+  event Debug(address from, bytes32 hash);
+
+  /**
+    * @notice Submit a presigned transfer
```

Comment:
It looks like it has the logic to prevent nonce reuse. But it is not immediately obvious how it works. 

Maybe a comment?

Reply:
Also do we have a test for nonce reuse attack?

Reply:
No tests yet, I will do that :+1: